KeyType Simplification (#2146)
dotnet/machinelearning: docs/code/IDataViewTypeSystem.md

In addition to its underlying type, a key type specifies:

> * A count value, between `1` and `ulong.MaxValue`, inclusive

Reviewer: At my first glance, I didn't know what does `inclusive` mean but then I realize how stupid I was after reading through the following paragraph. Maybe
```suggestion
* A count value, from `1` to `ulong.MaxValue`.
```
can help other naive users?

Reviewer 2: I do not like the suggested change as much as I might like. Chaning "between" to "from" is good. (Between suggests exclusion.) Removing "inclusive" is bad. What you think?

> Regardless of the count value, the representational value zero

Reviewer: Can we use `ulong.MaxValue` to denote NaN? It can further simplify the conversion between integers and keys. Note that conversion happens in type casting and reading C# native class as `IDataView`.
```suggestion
Regardless of the count value, the representational value zero
```
@TomFinley, what do you think?

Reviewer: My original idea comes from having same value for key value and its underlying representation. This way we can simplify the loading/writing value from/to C# data structure. But I got your point, NA'd better to be unique.

always means `NA` and the representational value one is always the first valid
value of the key type.

Notes:

* The `Count` property returns the count, or cardinality of the key type. This 
  is of type `ulong`. The legal representation values are from one up to 
  and including `Count`. The `Count` is required to be representable in the 
  underlying type, so, for example, the `Count` value of a key type based
  on `System.Byte` must not exceed `255`. As an example of the usefulness 
  of the `Count` property, consider the `KeyToVector` transform implemented
  as part of ML.NET. It maps from a key type value to an indicator vector.
  The length of the vector is the `Count` of the key type, which is required
  to be positive. For a key value of `k`, with `1 ≤ k ≤ Count`, the resulting

>   vector has a value of one in the (`k-1`)th slot, and zero in all other slots.

Reviewer: Maybe
```suggestion
  vector has a value of one in the (`k`)th slot, and zero in all other slots.
```
? The 1st slot is indexed by key value 1, isn't it?

Reviewer 2: Depends on whether for teh sake of this discussion we're considering the "first" slot to be literally 1st or 0th.

Reviewer: I see. `k-1` is fine, so let me close this comment.
